pyproject.toml

```diff
@@ -31,6 +31,7 @@ packages = [
 [tool.poetry.dependencies]
 python = "^3.6"
 graphql-core = "^3.0"
+typing-extensions = "^3.6.2"
```

Comment:
You should make this a [Python restricted dependency](https://python-poetry.org/docs/dependency-specification/#python-restricted-dependencies), only required for Py 3.6 and 3.7. Protocol is in the typing module since Py 3.8.

Also, the same dependency needs to be added in setup.py (currently we are supporting both, setup.py and pyproject.toml). There, you can use an [environment marker](https://www.python.org/dev/peps/pep-0508/#environment-markers) for the restricted dependency.